lib/libvfio-user.c

```diff
@@ -1052,6 +1052,45 @@ handle_dirty_pages(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
     return ret;
 }
 
+static int
+handle_device_feature(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
+{
+    struct vfio_user_device_feature *req = msg->in.iov.iov_base;
+
+    if (vfu_ctx->migration == NULL) {
+        return -EINVAL;
+    }
+
+    if (!migration_feature_supported(req->flags)) {
+        // FIXME what error code to return? we really want "not supported"
+        //   instead of "not permitted"?
+        return -EINVAL;
+    }
+
+    ssize_t ret;
+
+    if (req->flags & VFIO_DEVICE_FEATURE_PROBE) {
```

Comment:
presumably it's invalid to set any other flag its in this case; should you check for that?

Reply:
It's not invalid - you set `GET` and `SET` to specify which operations you are probing support for.

From the spec:
> `VFIO_DEVICE_FEATURE_PROBE` instructs the server to probe for feature support. If `VFIO_DEVICE_FEATURE_GET/SET` are also set, the probe will only return success if all of the indicated methods are supported.

Reply:
yeah, so I read after.